src/features/DebugSession.ts

```diff
@@ -42,6 +43,88 @@ export class DebugSessionFeature implements IFeature, DebugConfigurationProvider
         }));
     }
 
+    public async provideDebugConfigurations(
+        folder: WorkspaceFolder | undefined,
+        token?: CancellationToken): Promise<DebugConfiguration[]> {
+
+        const launchAttachItems = [
+            { label: "Launch",
+              description: "Launch the debugger with a specified script or for the interactive session" },
+            { label: "Attach",
+              description: "Attach the debugger to a running PowerShell Host Process" },
+        ];
+
+        const debugTypeSelection =
+            await vscode.window.showQuickPick(
+                launchAttachItems,
+                { placeHolder: "Would you like to launch or attach the PowerShell debugger?" });
+
+        let debugConfiguration = [];
+
+        if (debugTypeSelection.label === "Launch") {
```

Comment:
This logic here I would write as:

```typescript
const powerShellType = 'PowerShell';

if (debugTypeSelection.label !== "Launch") {
    return [
        {
            name: 'PowerShell: Attach to PowerShell Host Process',
            type: powerShellType,
            request: 'attach',
            runspaceId: 1
        }
    ]
}

const launchCurrentFileLabel = 'Launch Current File';
const launchScriptLabel = 'Launch Script';
const interactiveSessionLabel = 'Interactive Session';

const launchItems = [
    {
        label: launchCurrentFileLabel,
        description: 'Debugs the script currently in the active editor window'
    },
    {
        label: launchScriptLabel,
        description: 'Debugs the specified script'
    },
    {
        label: interactiveSessionLabel,
        description: 'Debugs scripts or modules executed from the Integrated Console'
    }
];

const launchSelection = await vscode.window.showQuickPick(
    launchItems,
    { placeHolder: 'Select a debugger launch option'});

if (launchSelection.label === launchCurrentFileLabel) {
    return [
        {
            name: 'PowerShell: Launch Current File',
            type: powerShellType,
            request: 'launch',
            script: '${file}',
            cwd: '${file}'
        }
    ]
}

if (launchSelection.label === launchScriptLabel) {
    return [
        {
            name: 'Powershell: Launch Script',
            type: powerShellType,
            request: 'launch',
            script: 'Enter path or command to execute, e.g.: ${workspaceFolder}/src/foo.ps1 or Invoke-Pester'
        }
    ]
}

// Last remaining possibility is interactive launch
return [
    {
        name: 'PowerShell: Interactive Session',
        type: powerShellType,
        request: 'launch',
        cwd: ''
    }
];
```

Reply:
If the impl remains  the way it is then yeah, it should be broken up into mutiple methods like you show above.  At this point, I'm more interested in feedback on the experience.  I think two levels of prompts is probably one too many.  Yesterday, I was thinking that perhaps the multi-selection approach would be better.  I'm really interested in getting @isidorn feedback because none of the other debuggers except for well, Python. It does prompt with this.  

![image](https://user-images.githubusercontent.com/5177512/61233950-0b63df80-a6ef-11e9-90b7-7fbc5882a23a.png)

BTW with that Python debug prompt above, if I press <kbd>escape</kbd> to cancel the operation, I get all the Python configs dumped into my launch.json.   That seems like a bug.